Content.Client/UserInterface/Controls/RadialContainer.cs

```diff
@@ -0,0 +1,107 @@
+    /// Both values must be between 0 and 2 PI radians
+    /// </summary>
+    /// <remarks>
+    /// The top of the screen is at 0 radians, and the bottom of the screen is at PI radians
+    /// </remarks>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public Vector2 AngularRange
+    {
+        get
+        {
+            return _angularRange;
+        }
+
+        set
+        {
+            var x = value.X;
+            var y = value.Y;
+
+            x = x > MathF.Tau ? x % MathF.Tau : x;
+            y = y > MathF.Tau ? y % MathF.Tau : y;
+
+            x = x < 0 ? MathF.Tau + x : x;
+            y = y < 0 ? MathF.Tau + y : y;
+
+            _angularRange = new Vector2(x, y);
+        }
+    }
+
+    private Vector2 _angularRange = new Vector2(0f, MathF.Tau - float.Epsilon);
+
+    /// <summary>
+    /// Determines the direction in which child elements will be arranged
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public RAlignment RadialAlignment { get; set; } = RAlignment.Clockwise;
+
+    /// <summary>
+    /// Determines how far from the radial container's center that its child elements will be placed
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float Radius { get; set; } = 100f;
+
+    /// <summary>
+    /// Sets whether the container should reserve a space on the layout for child which are not currently visible
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public bool ReserveSpaceForHiddenChildren { get; set; } = true;
+
+    /// <summary>
+    /// This container arranges its children, evenly separated, in a radial pattern
+    /// </summary>
+    public RadialContainer()
+    {
+
+    }
+
+    protected override void Draw(DrawingHandleScreen handle)
+    {
+        var children = ReserveSpaceForHiddenChildren ? Children : Children.Where(x => x.Visible);
+        var childCount = children.Count();
+
+        // Determine the size of the arc, accounting for clockwise and anti-clockwise arrangements
+        var arc = AngularRange.Y - AngularRange.X;
+        arc = (arc < 0) ? MathF.Tau + arc : arc;
+        arc = (RadialAlignment == RAlignment.AntiClockwise) ? MathF.Tau - arc : arc;
+
+        // Account for both circular arrangements and arc-based arrangements
+        var childMod = MathHelper.CloseTo(arc, MathF.Tau, 0.01f) ? 0 : 1;
+
+        // Determine the separation between child elements
+        var sepAngle = arc / (childCount - childMod);
+        sepAngle *= (RadialAlignment == RAlignment.AntiClockwise) ? -1f : 1f;
+
+        // Adjust the positions of all the child elements
+        for (int i = 0; i < childCount; i++)
+        {
+            var child = children.ElementAt(i);
+            var position = new Vector2(Radius * MathF.Sin(AngularRange.X + sepAngle * i) + Width / 2f - child.Width / 2f, -Radius * MathF.Cos(AngularRange.X + sepAngle * i) + Height / 2f - child.Height / 2f);
+
+            SetPosition(child, position);
+        }
+    }
```

Comment:
Is ElementAt the LINQ one? I think that has to re-iterate the collection again so you might be better off tracking the index yourself as it will be much faster.